### PR TITLE
Localnode interface

### PR DESCRIPTION
**Pull request by billfort**
### Proposed changes in this pull request
Explain the changes in this pull request in order to help the project maintainers understand the overall impact of it.

### Type (put an `x` where ever applicable)
- [ ] Bug fix: Link to the issue
- [x] Feature (Non-breaking change)
- [ ] Feature (Breaking change)
- [ ] Documentation Improvement

### Checklist
Please put an `x` against the checkboxes. Write a small comment explaining if its `N/A` (not applicable)

- [ ] Read the [CONTRIBUTION guidelines](https://github.com/nknorg/nkn#contributing).
- [ ] All the tests are passing after the introduction of new changes.
- [ ] Added tests respective to the part of code I have written.
- [ ] Added proper documentation where ever applicable (in code and README.md).
- [x] Code has been written according to [NKN-Golang-Style-Guide](https://github.com/nknorg/nkn/wiki/NKN-Golang-Style-Guide)

### Extra information
Any extra information related to this pull request.

### CODE REVIEW

**Comment by billfort** [comment]
1. change package nkn/v2/node to nkn/v2/lnode
2. create a new packge nkn/v2/node
3. create ILocalNode interface in nkn/v2/node, modify others' package to use ILocalNode.
4. move handler.go, remotenode.go, node.go into nkn/v2/node package.
5. de-couple other package from nkn/v2/lnode
6.

File: lnode/localnode.go [hunk 1]
```diff
@@ -80,13 +84,13 @@ func NewLocalNode(wallet *vault.Wallet, nn *nnet.NNet) (*LocalNode, error) {
 		TlsJsonRpcPort:     uint32(config.Parameters.HttpsJsonPort),
 	}
 
-	node, err := NewNode(nn.GetLocalNode().Node.Node, nodeData)
+	node1, err := node.NewNode(nn.GetLocalNode().Node.Node, nodeData)
```
Comment (yilunzhang) [inline lnode/localnode.go:87]: Suggest change `node1` to `n`

File: lnode/message.go [hunk 2]
```diff
@@ -1,10 +1,13 @@
-package node
+package lnode
 
 import (
 	"crypto/sha256"
 	"errors"
 	"fmt"
-	"time"
+
+	//	"time"
```
Comment (yilunzhang) [inline lnode/message.go:8]: Remove unused comments

File: node/remotenode.go [hunk 3]
```diff
@@ -8,14 +8,15 @@ import (
...
 )
 
 type RemoteNode struct {
 	*Node
-	localNode *LocalNode
-	nnetNode  *nnetnode.RemoteNode
-	sharedKey *[sharedKeySize]byte
+	localNode ILocalNode
+	NnetNode  *nnetnode.RemoteNode
+	SharedKey *[SharedKeySize]byte
```
Comment (yilunzhang) [inline node/remotenode.go:19]: Don't expose sharedKey unless absolutely necessary

File: lnode/message.go [hunk 4]
```diff
@@ -41,6 +44,7 @@ func (localNode *LocalNode) SerializeMessage(unsignedMsg *pb.UnsignedMessage, si
 	return proto.Marshal(signedMsg)
 }
 
+/*
```
Comment (yilunzhang) [inline lnode/message.go:47]: Do not submit used commented code

**Comment by yilunzhang** [comment]
A few suggestions in general:
1.  You can put information like the problem you are trying to solve, why you want to make the changes, what is being changed, etc in `Proposed changes in this pull request` section of the PR.
2. Use rebase to clean commit history. You can even squash everything into one commit if that's appropriate.
3. Use `gofmt` to format your code. Usually you can set your IDE to do it automatically on save.

File: node/remotenode.go (hunk 3 above)
Comment (billfort) [reply node/remotenode.go:19]: yes. I moved encryptMessage and decryptMessage function into node package. so remoteNode no need expose its sharedKey.

**Comment by billfort** [comment]
Yes, gofmt is set in IDE.

**Pull request closed without merging**